### PR TITLE
Include 'products' in category query

### PR DESCRIPTION
**Pull request by facundocapua**
### Description
As a Magento developer, I want to access the "products" directly from any category object so that it's easy for me to render products belonging to a category from that category object. 

**Acceptance criteria**
1. User can access products belonging to a category from that category object
1. There is a default limit on how many products are shown at a time so that performance is not hindered
1. User can access products beyond than those included in the default set by changing pagination or default pagesize
1. At least the following data is available: 
    * name
    * sku
    * product_type
    * description
    * short_description
    * media_gallery_entries (default medium or thumbnail image)
    * price (final calculated price)
    * review rating
    * custom_attribute (ex: featured, popular)
    * product_options
5. Changes are covered with web API functional tests. See [\Magento\GraphQl\Catalog\ProductViewTest::testQueryAllFieldsSimpleProduct](https://github.com/magento/magento2/blob/b3372064f574fd5aece0617009ee317db589febe/dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/ProductViewTest.php#L35) as an example.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/graphql-ce?pullRequest=48) <br/>All committers have signed the CLA.

File: app/code/Magento/CatalogGraphQl/etc/schema.graphqls [hunk 1]
```diff
@@ -404,6 +409,12 @@ type Products @doc(description: "The Products object is the top-level object ret
     filters: [LayerFilter] @doc(description: "Layered navigation filters array")
 }
 
+type CategoryProducts @doc(description: "The category products object returned in the Category query") {
+    items: [ProductInterface] @doc(description: "An array of products that match the specified search criteria")
```
Comment (paliarush) [inline app/code/Magento/CatalogGraphQl/etc/schema.graphqls]: Search criteria is not used to fetch category products list. Please correct the docs.

File: app/code/Magento/CatalogGraphQl/Model/Resolver/Category/Products.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\Resolver\ValueFactory;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Framework\GraphQl\Query\Resolver\Argument\SearchCriteria\Builder;
+use Magento\CatalogGraphQl\Model\Resolver\Products\Query\Filter;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+
+class Products implements ResolverInterface
```
Comment (paliarush) [inline app/code/Magento/CatalogGraphQl/Model/Resolver/Category/Products.php]: Please document the class.

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/CategoryTest.php [hunk 3]
```diff
@@ -109,4 +110,163 @@ public function testCategoriesTree()
...
+  }
+}
+QUERY;
+
+        $response = $this->graphQlQuery($query);
+        $this->assertArrayHasKey('products', $response['category']);
+        $this->assertArrayHasKey('total_count', $response['category']['products']);
+        $this->assertEquals(2, $response['category']['products']['total_count']);
+        $this->assertEquals(1, $response['category']['products']['page_info']['current_page']);
+        $this->assertEquals(20, $response['category']['products']['page_info']['page_size']);
```
Comment (paliarush) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/CategoryTest.php:271]: Please add assertions at least for the product attributes mentioned in the acceptance criteria. For one of the products should be enough.

**Review by paliarush — CHANGES_REQUESTED** [review]
@facundocapua thank you for contribution, please address a few comments and we should be good to merge the PR.

**Comment by paliarush** [comment]
- [ ] Please address build issue: https://travis-ci.com/magento/graphql-ce/jobs/125373510

**Comment by facundocapua** [comment]
@paliarush I have addressed all your suggestions.

Please take a look and let me know.

Thanks for your comments!

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/CategoryTest.php [hunk 4]
```diff
@@ -109,4 +112,306 @@ public function testCategoriesTree()
...
+                            'currency' => 'USD'
+                        ],
+                        'adjustments' => []
+                    ],
+                ]
+            ],
+            ['response_field' => 'sku', 'expected_value' => $product->getSku()],
+            ['response_field' => 'type_id', 'expected_value' => $product->getTypeId()],
+            ['response_field' => 'updated_at', 'expected_value' => $product->getUpdatedAt()],
+//            ['response_field' => 'weight', 'expected_value' => $product->getWeight()],
```
Comment (paliarush) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/CategoryTest.php]: Please remove comment

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/CategoryTest.php [hunk 5]
```diff
@@ -109,4 +112,306 @@ public function testCategoriesTree()
...
+            self::assertEquals(
+                $expectedValue,
+                $actualResponse[$responseField],
+                "Value of '{$responseField}' field in response does not match expected value: "
+                . var_export($expectedValue, true)
+            );
+        }
+    }
+
+    private function eavAttributesToGraphQlSchemaFieldTranslator($attributeCode)
```
Comment (paliarush) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/CategoryTest.php]: Unused method. Please remove.

**Comment by paliarush** [comment]
@facundocapua the test looks great now! Please address couple minor comments and then we can merge the PR.

**Comment by facundocapua** [comment]
@paliarush Thanks for you review!! I have removed unused code.

**Review by paliarush — APPROVED** [review]

**Review by misha-kotov — APPROVED** [review]
Approved

**Comment by paliarush** [comment]
@facundocapua, please see if you can reproduce the failure which we have on our CI environment when running the whole web API test suite:

```
Magento\GraphQl\Catalog\CategoryTest::testCategoryProducts
Failed asserting that 1 matches expected 2.

/.../dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/CategoryTest.php:269
```

**Comment by paliarush** [comment]
@vgelani thank you for merging the PR, however it cannot be delivered to `magento/magento2` repository yet because of failing test. It will be delivered after the test is fixed.

**Comment by paliarush** [comment]
PR delivered to mainline. @facundocapua thank you for contribution!

**Pull request merged**